intent-classifier/training/train_tagger.js

```diff
@@ -0,0 +1,278 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+const fs = require('fs');
+const path = require('path');
+
+const ndjson = require('ndjson')
+const argparse = require('argparse');
+const tf = require('@tensorflow/tfjs');
+const fileIO = require('@tensorflow/tfjs-node/dist/io/file_system');
+const mkdirp = require('mkdirp');
+
+const {TAGS} = require('./util');
+
+const {getModel} = require('./tagger_model');
+
+const EMBEDDING_SIZE = 512;
+
+async function loadNDJSON(path) {
+  const records = [];
+
+  const parser = fs.createReadStream(path).pipe(ndjson.parse());
+
+  parser.on('data', obj => records.push(obj));
+
+  return new Promise(function(resolve, reject) {
+    parser.on('end', () => resolve(records));
+    parser.on('error', e => reject(e));
+  });
+}
+
+/**
+ * Returns a generator function that yields batches of data for training.
+ *
+ * @param {string} embeddingsPath path to ndjson file with token embeddings
+ * @param {string} taggedTokensPath path to ndjson with tagged tokens
+ */
+async function getDataIterator(
+    embeddingsPath, taggedTokensPath, sequenceLength, batchSize) {
```

Comment:
file a separate issue to switch this file to using tf.data once the other issue is resolved